wdl-engine/src/eval/v1/task.rs

```diff
@@ -0,0 +1,929 @@
+                    uri = document.uri()
+                );
+
+                // Tasks only have a root scope (0), an output scope (1), and a `task` variable
+                // scope (2)
+                let mut scopes = [
+                    Scope::new(None),
+                    Scope::new(Some(ROOT_SCOPE_INDEX.into())),
+                    Scope::new(Some(OUTPUT_SCOPE_INDEX.into())),
+                ];
+                let mut command = String::new();
+                let mut requirements = None;
+                let mut hints = None;
+
+                let nodes = toposort(&graph, None).expect("graph should be acyclic");
+                let mut current = 0;
+                while current < nodes.len() {
+                    match &graph[nodes[current]] {
+                        TaskGraphNode::Input(decl) => {
+                            self.evaluate_input(
+                                document,
+                                execution.as_ref(),
+                                &mut scopes,
+                                task,
+                                decl,
+                                inputs,
+                            )?;
+                        }
+                        TaskGraphNode::Decl(decl) => {
+                            self.evaluate_decl(
+                                document,
+                                execution.as_ref(),
+                                &mut scopes,
+                                task,
+                                decl,
+                            )?;
+                        }
+                        TaskGraphNode::Output(_) => {
+                            // Stop at the first output; at this point the task can be executed
+                            break;
+                        }
+                        TaskGraphNode::Command(section) => {
+                            assert!(command.is_empty());
+
+                            // Get the execution constraints
+                            let empty = Default::default();
+                            let constraints = execution
+                                .constraints(
+                                    self.engine,
+                                    requirements.as_ref().unwrap_or(&empty),
+                                    hints.as_ref().unwrap_or(&empty),
+                                )
+                                .with_context(|| {
+                                    format!("failed to execute task `{task}`", task = task.name())
+                                })?;
+
+                            // Introduce the task variable at this point; valid for both the command
+                            // section and the outputs section
+                            if version >= SupportedVersion::V1(V1::Two) {
+                                let task =
+                                    TaskValue::new_v1(task.name(), "bar", &definition, constraints);
```

Comment:
"bar"?

Reply:
Oh whoops, I forgot to fix the placeholder for the task id; I'll correct that.